### DISCUSSION TITLE
Can I use key pair authentication method for snowflake?

### DISCUSSION THREAD

**Discussion by hrkachhia** [theory/sqitch#756] [Q&A]
Hello Support, 

currently, I am using sqitch docker with providing user name and password in config file. Now, I want to use key pair authentication method instead for deployment purpose. Can you please guide me how to do this with sqitch? I want to pass only key instead of username and password like git. Is it possible with docker sqitch?  Please guide me.

**Comment by theory** [comment]
In the sqitch-authentication guide, go to the [Don't use Passwords](http://sqitch.org/docs/manual/sqitch-authentication/#dont-use-passwords) section and scroll down to the "Snowflake" bullet, which describes how to use a key pair. If you're using the [shell script](https://github.com/sqitchers/docker-sqitch/blob/main/docker-sqitch.sh), it will mount your the directory on your host machine to read the `~/.snowsql/config` file and copy the `SNOWSQL_PRIVATE_KEY_PASSPHRASE` environment variable into the container.
